test/directives/vector_vectorlength_num_fixed.f90

```diff
@@ -0,0 +1,33 @@
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+! Check that the vector vectorlength(fixed) directive generates the correct metadata.
+! RUN: %flang -O0 -S -emit-llvm %s -o - | FileCheck %s --check-prefixes=CHECK-00,CHECK-ALL
+
+! Check that "-Hx,59,2" disables vector directive.
+! RUN: %flang -Hx,59,2 -S -emit-llvm %s -o - | FileCheck %s --check-prefixes=CHECK-DISABLED,CHECK-ALL
+
+subroutine func1(a, b, m)
+! CHECK-ALL: define void @func1
+  integer :: i, m, a(m), b(m)
+  !dir$ vector vectorlength(2,fixed)
```

Comment:
Small point, but let's mix up the integer value you use in the tests. Using 2 across the board would make the tests vulnerable to a regression where 2 is always put here. Suggest using 2 in one of the tests, and use some of the other values across the other tests for num.